src/SpotifyAudioSource/SpotifyAudioSource.cs

```diff
@@ -645,16 +684,32 @@ private void OnSettingChanged(string settingName)
 
         private async Task RefreshAccessToken()
         {
+            if (string.IsNullOrWhiteSpace(RefreshToken))
+            {
+                Logger.Warn("RefreshToken missing. Please restart and go through the authorization process again.");

```

Comment:
Restart the app, the auth process or the whole computer?

Reply:
It's the Authorization that needs to be redone, but authorization flow is triggered on app start and also when you select Spotify as an active AudioSource, so either restarting the app or Deselecting then selecting Spotify as an audioSource should work. I think the Auth flow is also triggered when you save the spotify settings, like if you would change your secret key for example, so i guess that would also work